google/cloud/aiplatform/training_jobs.py

```diff
@@ -1753,3 +1754,329 @@ def _model_upload_fail_string(self) -> str:
             f"Training Pipeline {self.resource_name} is not configured to upload a "
             "Model."
         )
+
+
+class AutoMLImageTrainingJob(_TrainingJob):
+    def __init__(
+        self,
+        display_name: str,
+        prediction_type: str = "classification",
+        multi_label: bool = False,
+        model_type: str = "CLOUD",
+        base_model: Optional[models.Model] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Constructs a AutoML Image Training Job.
+
+        Args:
+            display_name (str):
+                Required. The user-defined name of this TrainingPipeline.
+            prediction_type (str):
+                The type of prediction the Model is to produce, one of:
+                    "classification" - Predict one out of multiple target values is
+                        picked for each row.
+                    "object_detection" - Predict a value based on its relation to other values.
+                        This type is available only to columns that contain
+                        semantically numeric values, i.e. integers or floating
+                        point number, even if stored as e.g. strings.
+            multi_label: bool = False
+                Required. Default is False.
+                If false, a single-label (multi-class) Model will be trained
+                (i.e. assuming that for each image just up to one annotation may be
+                applicable). If true, a multi-label Model will be trained (i.e.
+                assuming that for each image multiple annotations may be applicable).
+            model_type: str = "CLOUD"
```

Comment:
but the acceptable model_types depends on prediction_type, and this is validated in the function body below.

would it be cleaner if the public class is specialized, such as `AutoMLImageClassificationTrainingJob`, etc.?

Reply:
Agreed it would be cleaner, but after checking with @sasha-gitg the plan is to get this class out then released the specialized down the road.